imageproxy.go

```diff
@@ -189,6 +194,13 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 	if p.FollowRedirects {
 		// FollowRedirects is true (default), ensure that the redirected host is allowed
 		p.Client.CheckRedirect = func(newreq *http.Request, via []*http.Request) error {
+			if p.MaxRedirects > 0 && uint8(len(via)) > p.MaxRedirects {
+				if p.Verbose {
+					p.logf("followed too many redirects: %d", len(via))
+				}
+				http.Error(w, errTooManyRedirects.Error(), http.StatusBadRequest)
```

Comment:
I really don't know what good response status is for this scenario so I just went for the catch-all `400` 🤷‍♂️

Reply:
you shouldn't need to call `http.Error` at all.  For example, look at the [defaultCheckRedirect](https://cs.opensource.google/go/go/+/master:src/net/http/client.go;l=811-816?q=redirect&ss=go%2Fgo:src%2Fnet%2Fhttp%2F), which simply returns an error.

Reply:
I'm not sure if it's a good implementation but I've come up with [this](https://github.com/willnorris/imageproxy/pull/324/commits/c4af062fea5f1e5e54d12e2afc3f6d10a5408a63) in order to avoid calling `http.Error(...)` explicitly